src/textAngularSetup.js

```diff
@@ -565,16 +565,30 @@ angular.module('textAngularSetup', [])
 			var urlPrompt;
 			urlPrompt = $window.prompt(taTranslations.insertVideo.dialogPrompt, 'https://');
 			if (urlPrompt && urlPrompt !== '' && urlPrompt !== 'https://') {
+				var videoId = null;
 				// get the video ID
-				var ids = urlPrompt.match(/(\?|&)v=[^&]*/);
```

Comment:
Instead of having to create if cases for each version of youtbe url variatio you could do somthing like
```javascript
var videoId = null;
var re = new RegExp('(?:youtube(?:-nocookie)?\.com/(?:[^/]+/.+/|(?:v|e(?:mbed)?)/|.*[?&]v=)|youtu\.be/)([^"&?/ ]{11})','i');
matches = urlPrompt.match(re):
if (match) {
  videoId = match[0]
}
````
This should handle:
```
http://www.youtube.com/watch?v=0zM3nApSvMg&feature=feedrec_grec_index
http://www.youtube.com/user/IngridMichaelsonVEVO#p/a/u/1/QdK8U-VIH_o
http://www.youtube.com/v/0zM3nApSvMg?fs=1&amp;hl=en_US&amp;rel=0
http://www.youtube.com/watch?v=0zM3nApSvMg#t=0m10s
http://www.youtube.com/embed/0zM3nApSvMg?rel=0
http://www.youtube.com/watch?v=0zM3nApSvMg
http://youtu.be/0zM3nApSvMg
```